From a725cb4159746a006538e2b8e13d164339ef9135 Mon Sep 17 00:00:00 2001
From: Robin Gong <b38343@freescale.com>
Date: Sun, 4 Jan 2015 15:50:23 +0800
Subject: [PATCH 1149/1543] MLK-10073-1 ARM: dts: imx6sx-sdb: add
 imx6sx-sdb-revb board support

commit c9d110ef57ea852c813188e818143cf0463a2bdb from
git://git.freescale.com/imx/linux-2.6-imx.git

Add reva board dts file since we use revb board as the default dts file.

Signed-off-by: Robin Gong <b38343@freescale.com>
---
 arch/arm/boot/dts/Makefile            |    1 +
 arch/arm/boot/dts/imx6sx-sdb-reva.dts |   32 ++++++++++++++++++++++++++++++++
 arch/arm/boot/dts/imx6sx-sdb.dts      |   15 +--------------
 3 files changed, 34 insertions(+), 14 deletions(-)
 create mode 100644 arch/arm/boot/dts/imx6sx-sdb-reva.dts

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index 3c97b42..5bbbb65 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -196,6 +196,7 @@ dtb-$(CONFIG_ARCH_MXC) += \
 	imx6sx-19x19-arm2-lcdif1.dtb \
 	imx6sx-19x19-arm2-gpmi-weim.dtb \
 	imx6sx-sdb.dtb \
+	imx6sx-sdb-reva.dtb \
 	imx6sx-sabreauto.dtb \
 	imx6sx-sabreauto-m4.dtb \
 	imx6sx-sdb-lcdif1.dtb \
diff --git a/arch/arm/boot/dts/imx6sx-sdb-reva.dts b/arch/arm/boot/dts/imx6sx-sdb-reva.dts
new file mode 100644
index 0000000..13a9953
--- /dev/null
+++ b/arch/arm/boot/dts/imx6sx-sdb-reva.dts
@@ -0,0 +1,32 @@
+/*
+ * Copyright (C) 2015 Freescale Semiconductor, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include "imx6sx-sdb.dts"
+
+/*
+ * The CAN transceiver is changed on RevB board.
+ * This dts is used for support the old CAN transceiver on RevA board
+ */
+
+&pmic {
+	compatible = "fsl,pfuze100";
+
+	regulators {
+		sw1c_reg: sw1c {
+			regulator-min-microvolt = <300000>;
+			regulator-max-microvolt = <1875000>;
+			regulator-boot-on;
+			regulator-always-on;
+			regulator-ramp-delay = <6250>;
+		};
+		sw4_reg: sw4 {
+			regulator-min-microvolt = <800000>;
+			regulator-max-microvolt = <3300000>;
+		};
+	};
+};
diff --git a/arch/arm/boot/dts/imx6sx-sdb.dts b/arch/arm/boot/dts/imx6sx-sdb.dts
index 35cd9a5..46cbc7a 100644
--- a/arch/arm/boot/dts/imx6sx-sdb.dts
+++ b/arch/arm/boot/dts/imx6sx-sdb.dts
@@ -459,7 +459,7 @@
 	status = "okay";
 
 	pmic: pfuze100@08 {
-		compatible = "fsl,pfuze100";
+		compatible = "fsl,pfuze200";
 		reg = <0x08>;
 
 		regulators {
@@ -471,14 +471,6 @@
 				regulator-ramp-delay = <6250>;
 			};
 
-			sw1c_reg: sw1c {
-				regulator-min-microvolt = <300000>;
-				regulator-max-microvolt = <1875000>;
-				regulator-boot-on;
-				regulator-always-on;
-				regulator-ramp-delay = <6250>;
-			};
-
 			sw2_reg: sw2 {
 				regulator-min-microvolt = <800000>;
 				regulator-max-microvolt = <3300000>;
@@ -500,11 +492,6 @@
 				regulator-always-on;
 			};
 
-			sw4_reg: sw4 {
-				regulator-min-microvolt = <800000>;
-				regulator-max-microvolt = <3300000>;
-			};
-
 			swbst_reg: swbst {
 				regulator-min-microvolt = <5000000>;
 				regulator-max-microvolt = <5150000>;
-- 
1.7.5.4

